graphbolt/src/csc_sampling_graph.cc

```diff
@@ -222,28 +222,27 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighborsImpl(
     const torch::Tensor& nodes, bool return_eids, NumPickFn num_pick_fn,
     PickFn pick_fn) const {
   const int64_t num_nodes = nodes.size(0);
-  const int64_t num_threads = torch::get_num_threads();
-  std::vector<torch::Tensor> picked_neighbors_per_thread(num_threads);
+  const auto indptr_options = indptr_.options();
   torch::Tensor num_picked_neighbors_per_node =
-      torch::zeros({num_nodes + 1}, indptr_.options());
+      torch::zeros({num_nodes + 1}, indptr_options);
```

Comment:
We better use torch::empty here. Every index except 0 is already overwritten in the num_pick_fn. So we simply set the 0 index to zero manually.